src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -54,6 +54,13 @@
     private static final List<String> ALWAYS_MATCHING_CONTEXT_STRATEGY_CHECKS =
             Arrays.asList("OuterTypeNumberCheck");
 
+    /**
+     * List of checks that support context strategy but need modify violation nodes
+     * to their parent abstract nodes to get their child nodes.
+     */
+    private final List<String> abstractNodeChecks =
+            Arrays.asList("ParameterNumberCheck");
```

Comment:
remove this default value.
No names in code.

All names from code above .... is unfortunate leak.

Reply:
@romani ok, I will remove it, then if user use ParameterNumberCheck, they need to config `<property name="checkNameForContextStrategyByTokenOrParent" value="ParameterNumber," />` by themselves, it is ok, right?

Reply:
Yes, all should come from config. At the very end we might put some names  by default values. But for now no default, we are investigating ...

Reply:
ok, done